### PR TITLE
analog: add DataSegmentsCollector, frame

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/utils/BinaryBuffer.ts [hunk 1]
```diff
@@ -216,6 +216,7 @@ export interface IBinaryBuffer {
     bytesLeft: number,
     position: number,
 
+    toUint8Array (): types.TBytes,
```
Comment (Pnom) [inline src/utils/BinaryBuffer.ts:219]: line break?

File: src/analog/utils/frame.ts [hunk 2]
```diff
@@ -1,50 +1,53 @@
...
+    message: IMessage | IInvalidMessage
 }
 
 
-export const getCommands = ( {frame, message}: IAnalogFrame, isStrict: boolean = false ): Array<Command> => {
-    if ( isStrict && !frame.isValid ) {
-        return [];
-    }
+// export const getCommands = ( {frame, message}: IAnalogFrame, isStrict: boolean = false ): Array<ICommand> => {
+//     if ( isStrict && !frame.isValid ) {
```
Comment (Pnom) [inline src/analog/utils/frame.ts:18]: no need to comment, remove

File: src/utils/BinaryBuffer.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/utils/BinaryBuffer.ts:219]: added

File: src/analog/utils/frame.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/analog/utils/frame.ts:18]: removed

**Pull request merged**